server/src/lib/webhooks/utils.ts

```diff
@@ -0,0 +1,29 @@
+import crypto from 'crypto';
+import { Request } from 'express';
+
+export const verifyWebhookSecret = (req: Request) => {
+  // we get signature from gh
+  const signature = req.header('X-Hub-Signature-256');
+  // we encrypt signature from our db
+
+  // we take repo id
+  const repoId = req.body.repository.id.toString();
+
+  // we take first 4 chars from gh client id
+  const clientIDPart = process.env.GITHUB_CLIENT_ID.slice(0, 4);
```

Comment:
I had some thought about this one, and if the secret of one repo leak you can guess the other secrets which is not good..
What do you think about adding a new field to the app database schema and generate a random secret during the app creation? Then we can show it to the user in the settings

Reply:
Yeah, this sounds best.